python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py

```diff
@@ -454,18 +498,33 @@ def predict_interval(self, data, val_data=None, batch_size=32, repetition_times=
         # step1, according to validation dataset, calculate inherent noise
         # which should be done during fit
         if not hasattr(self, "data_noise"):
-            invalidInputError(val_data is not None,
+            invalidInputError(validation_data is not None,
                               "When call predict_interval for the first time, you must pass in "
-                              "validation data to calculate data noise.")
+                              "validation_data to calculate data noise.")
+            # transform a TSDataset instance to dataloader
+            if isinstance(validation_data, TSDataset):
+                _rolled = validation_data.numpy_x is None
+                validation_data = validation_data.to_torch_data_loader(
+                    batch_size=batch_size,
+                    roll=_rolled,
+                    lookback=self.data_config['past_seq_len'],
+                    horizon=self.data_config['future_seq_len'],
+                    label_len=self.data_config['label_len'],
+                    time_enc=True,
+                    feature_col=data.roll_feature,
+                    target_col=data.roll_target,
+                    shuffle=False
+                )
             # data transform
-            if isinstance(val_data, DataLoader):
-                target = np.concatenate(tuple(val[1] for val in val_data), axis=0)
+            if isinstance(validation_data, DataLoader):
+                target = np.concatenate(tuple(val[1] for val in validation_data),
+                                        axis=0)
             else:
-                x, target, x_enc, y_enc = val_data
+                _, target, _, _ = validation_data
 
-            target = target[:, -self.internal.pred_len:, :]
+            target = target[:, -self.data_config['future_seq_len']:, :]
```

Comment:
any particular reason to change this one?

Reply:
Just to align with the previous code